lib/srv/ingress/reporter.go

```diff
@@ -76,16 +79,34 @@ var (
 	authenticatedConnectionsAccepted = prometheus.NewCounterVec(prometheus.CounterOpts{
 		Namespace: "teleport",
 		Name:      "authenticated_accepted_connections_total",
-	}, commonLabels)
+	}, []string{"ingress_path", "ingress_service", "service_metadata"})
```

Comment:
Initially I thought that `authenticatedConnectionsAccepted` metrics is used only in Database access but it look like it shared between other services. 

Thus in this approach we are artificially trying to extend the    `authenticatedConnectionsAccepted` by using the `service_metadata` metadata label that won't allow to fully leverage label grouping (service_metadata is string combined with the`;` char).  I think that the better approach we be just introduce two additional databases labels for connection count with proper metrics tags. 

@Tener 
Does this sounds resonable ?

Reply:
This is one of the options, sure. I'm open to this one as well, although I'm worried additional metrics may end up being more confusing than helpful. @dboslee what would you say is the preferred solution in this case?

Reply:
As discussed offline, I've changed this PR toward the middle ground: the metrics are still shared, but instead of concatenating heterogenous values, there are now three discrete variable-meaning labels. I've also dropped kube/db/app names from the TLS-based metrics: there can be thousands of different services in production settings.

Reply:
I agree with @smallinsky here. stuffing a bunch of data into a single label will make it difficult to use that data for querying. It also seems wrong to have database specific labels for these metrics.

What are we trying to measure with this? It seems like we want to see where a connection is being proxied to.

I could see something like a `destination` label being applicable across all the services monitored here but if not it might make sense to have separate metrics to count proxied connections to different resources.

Reply:
@dboslee I have refactored the PR slightly; there are three distinct labels now, their meaning dependent on the service used. Do you think this is a viable way, or should I scrap this approach entirely and go with custom metrics instead?

Reply:
I would scrap the idea of `metadata` and go with something more concrete. I think a `destination` label could work here which would categorize what type of resource the connection is being proxied to. Is there something else you are trying to capture with the labels? Maybe a totally separate metric would make sense in that case.

Reply:
All right, I'll work on a different metric with a more narrow scope. Thanks for the feedback!